### PR TITLE
Initialize the size of each Arrow builder based on previous batches

### PR DESCRIPTION
**Pull request by lquerel**
Initialize the size of each Arrow builder based on previous batches size to minimize the number of reallocation.
+ documentation, code cleanup, and more linters.

### CODE REVIEW

File: pkg/otel/common/arrow/schema.go [hunk 1]
```diff
@@ -282,6 +318,100 @@ func (m *AdaptiveSchema) rebuildSchema(updates []SchemaUpdate) {
...
+		}
+	case *array.Map:
+		keyField := field.Type.(*arrow.MapType).KeyField()
+		valueField := field.Type.(*arrow.MapType).ItemField()
+		m.collectSizeBuildersFromArray(path+".key", &keyField, arr.Keys())
+		m.collectSizeBuildersFromArray(path+".value", &valueField, arr.Items())
+	}
+}
+
+// initSizeBuildersFromRecordBuilder initializes the size of each internal builder in rhe record builder
```
Comment (jmacd) [inline pkg/otel/common/arrow/schema.go:367]:
```suggestion
// initSizeBuildersFromRecordBuilder initializes the size of each internal builder in the record builder
```

File: pkg/otel/common/arrow/schema.go [hunk 2]
```diff
@@ -282,6 +318,100 @@ func (m *AdaptiveSchema) rebuildSchema(updates []SchemaUpdate) {
...
+	for i, arr := range arrays {
+		field := schema.Field(i)
+		m.collectSizeBuildersFromArray(field.Name, &field, arr)
+	}
+}
+
+func (m *AdaptiveSchema) collectSizeBuildersFromArray(path string, field *arrow.Field, arr arrow.Array) {
+	window, found := m.fieldCapacities[path]
+	if !found {
+		window = NewBuilderCapacityWindow(10)
```
Comment (jmacd) [inline pkg/otel/common/arrow/schema.go:336]: Suggest adding `const builderCapacityWindowSize = 10` at the top with a comment and using it here.

**Review by jmacd — APPROVED** [review]

File: SECURITY.md [hunk 3]
```diff
@@ -14,3 +14,7 @@ at F5SIRT@f5.com. Mention OTEL Arrow Adapter in the subject line.
 
 For more information visit
 [https://www.f5.com/services/support/report-a-vulnerability](https://www.f5.com/services/support/report-a-vulnerability)
+
+## Threat Model
+
+See the following [threat model assessment](docs/thread_model_assessment.md) for more details.
```
[the remark this answers is not in the document]
Comment (lquerel) [reply SECURITY.md:20]: Fixed

File: pkg/otel/common/arrow/schema.go (hunk 2 above)
Comment (lquerel) [reply pkg/otel/common/arrow/schema.go]: Done

**Pull request merged**